pkg/scanner/scanner.go

```diff
@@ -86,6 +108,63 @@ func (s *scannerService) Watch(opts ...ScanRequestOption) {
 	s.requestsQueue.enqueue(req)
 }
 
+func (s *scannerService) WatchDescriptorWallet(
+	desc string,
+	eventType []EventType,
+	blockStart int,
+) error {
+	for _, v := range eventType {
+		switch v {
+		case UnspentUtxo:
+			wallet, err := descriptor.Parse(desc)
+			if err != nil {
+				return err
+			}
+
+			var scripts []descriptor.ScriptResponse
+
+			if wallet.IsRange() {
+				scripts, err = wallet.Script(descriptor.WithRange(numOsScripts))
+				if err != nil {
+					return err
+				}
+
+				for _, v := range scripts {
+					s.Watch(
+						WithStartBlock(uint32(blockStart)),
+						WithWatchItem(&ScriptWatchItem{
+							outputScript: v.Script,
+						}),
+						WithPersistentWatch(),
+					)
+					if err != nil {
```

Comment:
it seems err is not assigned in for loop ?

Reply:
Done